src/drivers/next-json-cookie/index.ts

```diff
@@ -0,0 +1,78 @@
+import nextCookies from 'cookies-next';
+import { CookieSerializeOptions } from 'cookie';
+import AES from 'crypto-js/aes';
+import Utf8 from 'crypto-js/enc-utf8';
+import CacheDriver from '../driver';
+import { CookieValueTypes, OptionsType } from 'cookies-next/lib/types';
+import DEFAULT_COOKIE_OPTIONS from './default-cookie-options';
+import { decryptCookie, encryptCookie } from './encrypt-cookie';
+import JSONValue from '../types/json';
+
+const CACHE_MANAGER_COOKIE_NAMESPACE = 'hs_cache_'
+
+interface Config extends CookieSerializeOptions {
+  crypto?: typeof AES,
+  encoder?: typeof Utf8,
+}
+
+class NextCookieDriver<TmpCookiesObj> extends CacheDriver<typeof nextCookies> {
+
+  #config?: CookieSerializeOptions;
+  #crypto?: typeof AES;
+  #encoder?: typeof Utf8;
+
+  constructor(protected store: typeof nextCookies, { crypto, encoder, ...config }: Config = {}) {
+    super();
+
+    this.#config = config;
+    this.#crypto = crypto;
+    this.#encoder = encoder;
+  }
```

Comment:
The prefix should be in the config as well, as we may want to manage cookies for different features, for example we may have a cookie cache manager on `hs-next` for `hs-enquire` related functionality, and another for `hs-360` related functionality, when we call `flush()` I'd like to feel confident that the cookies I'm clearing will be scoped to one or the other.

```suggestion
interface Config extends CookieSerializeOptions {
  crypto?: typeof AES,
  encoder?: typeof Utf8,
  prefix?: string,
}

class NextCookieDriver<TmpCookiesObj> extends CacheDriver<typeof nextCookies> {

  #config?: CookieSerializeOptions;
  #crypto?: typeof AES;
  #encoder?: typeof Utf8;
  #prefix?: string;

  constructor(protected store: typeof nextCookies, config: Config = {}) {
    super();

    const { 
      crypto, 
      encoder, 
      prefix = 'hs_cache_', 
      ...cookieOptions 
    } = config;

    this.#config = cookieOptions;
    this.#crypto = crypto;
    this.#encoder = encoder;
    this.#prefix = prefix;
  }
```